go/cli/mcap/cmd/cat.go

```diff
@@ -109,57 +110,78 @@ func printMessages(
 			}
 			continue
 		}
-		switch schema.Encoding {
-		case "ros1msg":
-			transcoder, ok := transcoders[channel.SchemaID]
-			if !ok {
-				packageName := strings.Split(schema.Name, "/")[0]
-				transcoder, err = ros.NewJSONTranscoder(packageName, schema.Data)
+		if schema == nil || schema.Encoding == "" {
```

Comment:
@jon-chuang  is the idea with this patch to split out the change for supporting nil schemas, from support for CBOR? It seems like it still depends on the CBOR change, so I don't understand how it's different from the other patch.

Reply:
Sorry, this PR is building on the previous patch. The main change is handling unrecognized encoding by writing the raw data bytes insted of throwing an error. I'll convert this to a draft, it should be clearer once that PR is merged.